File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        Arrays.sort(nums);
        for(int num : nums) set.add(num);
        int max = 0;
        for(int i = 0; i < nums.length; i++) {
            if(!set.contains(nums[i] - 1)) { //if set doesnt contain prev number, then the sequence can only start here
                int curr = 1;
                for(int j = i + 1; j < nums.length; j++) {
                    if(nums[j] == nums[j-1]+1) curr++;
                    else if(nums[j] == nums[j-1]) continue;
                    else if(nums[j] > nums[j-1]+1) break;
                }
                max = Math.max(curr, max);
            } else {
                continue;
            }
        }
        return max;
        
    }
}

File: Data Structures & Algorithms/house-robber-ii/submission-1.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        res1, res2 = [], []
        if len(nums) <= 3:
            return max(nums)

        for i in range(len(nums)-1):
            if i == 0 or i == 1:
                res1.append(nums[i])
            elif i == 2:
                res1.append(max(nums[i], nums[i]+res1[i-2]))
            else:
                res1.append(max(nums[i]+res1[i-3], nums[i]+res1[i-2]))


        for i in range(1, len(nums)):
            if i == 1 or i == 2:
                res2.append(nums[i])
            elif i == 3:
                res2.append(max(nums[i], nums[i]+res2[i-3]))
            else:
                res2.append(max(nums[i]+res2[i-4], nums[i]+res2[i-3]))
        print(res1, res2)
        return max(max(res1), max(res2))